test_runner/src/main/kotlin/ftl/client/google/GcStorage.kt

```diff
@@ -114,6 +111,37 @@ object GcStorage {
         )
     }
 
+    private fun File.uploadWithProgress(
+        bucket: String,
+        path: String,
+        name: String
+    ) {
+        // TODO #2136 handle messages with state
+        runWithProgress(
+            startMessage = "Uploading file [$name] to ${GCS_STORAGE_LINK + join(bucket, path).replace(name, "")}..",
```

Comment:
I looked into the integration test failures. The failure that I'm seeing looks like:

```
at app//utils.IntergrationTestsUtilsKt.assertContainsUploads(IntergrationTestsUtils.kt:60)
at app//integration.MultipleApksIT.flank full option run(MultipleApksIT.kt:45)
```

The definition of `assertContainsUploads`  looks like: https://github.com/Flank/flank/blob/1cec01d5c0e22caec0e85d5702c77154b1cef12e/integration_tests/src/test/kotlin/utils/IntergrationTestsUtils.kt#L59-L61

This new `startMessage` doesn't match the assertion. Removing the word 'file' and following space results in passing tests.

It's a brittle test, but I don't know the details of why it was written that way. My recommendation would be to change the value of `startMessage` to match the assertion, and at some point in the future, follow up on making the test less brittle.

Reply:
I tired updating the test but for some reason `./gradlew integrationTests` doesn't seem to run on my local machine. When I run the `MultipleApksIT` tests I pass so I believe the fix worked

Reply:
I totally forgot that the integration tests require a GCP credential. I'm actually seeing a failure on two tests including the `MultipleApksIT` test.

I think what's happening is the string that gets passed into the pattern you introduced has regex reserved characters in it, and so the pattern is invalid.

```
integration.MultipleApksIT > flank full option run FAILED
    java.util.regex.PatternSyntaxException: Illegal character range near index 23
    Uploading( file)? [app-multiple-success-debug-androidTest.apk]
                           ^
        at java.base/java.util.regex.Pattern.error(Pattern.java:2027)
        at java.base/java.util.regex.Pattern.range(Pattern.java:2825)
        at java.base/java.util.regex.Pattern.clazz(Pattern.java:2713)
        at java.base/java.util.regex.Pattern.sequence(Pattern.java:2138)
        at java.base/java.util.regex.Pattern.expr(Pattern.java:2068)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1782)
        at java.base/java.util.regex.Pattern.<init>(Pattern.java:1428)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1094)
        at utils.IntergrationTestsUtilsKt.assertContainsUploads(IntergrationTestsUtils.kt:60)
        at integration.MultipleApksIT.flank full option run(MultipleApksIT.kt:45)

integration.CustomShardingIT > flank custom sharding -- android FAILED
    java.util.regex.PatternSyntaxException: Illegal character range near index 23
    Uploading( file)? [app-multiple-success-debug-androidTest.apk]
                           ^
        at java.base/java.util.regex.Pattern.error(Pattern.java:2027)
        at java.base/java.util.regex.Pattern.range(Pattern.java:2825)
        at java.base/java.util.regex.Pattern.clazz(Pattern.java:2713)
        at java.base/java.util.regex.Pattern.sequence(Pattern.java:2138)
        at java.base/java.util.regex.Pattern.expr(Pattern.java:2068)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1782)
        at java.base/java.util.regex.Pattern.<init>(Pattern.java:1428)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1094)
        at utils.IntergrationTestsUtilsKt.assertContainsUploads(IntergrationTestsUtils.kt:60)
        at integration.CustomShardingIT.flank custom sharding -- android(CustomShardingIT.kt:73)
```